tests/D2L.CodeStyle.Analyzers.Test/UnsafeStatics/UnsafeStaticsAnalyzerTests.cs

```diff
@@ -672,6 +674,61 @@ internal class Bar {
 			AssertSingleDiagnostic( test, 7, 41, "foo", "test.Tests.Foo" );
 		}
 
+		[Test]
+		public void DocumentWithStaticField_ReadonlyUnsafeBaseClassWithUnsafeInitializer_Diag() {
+			const string test = @"
+	using System;
+	namespace test {
+		class Tests {
+			interface IUnsafe { void Magic(); } // could be anythinggggggg
+
+			class Safe : IUnsafe {
+				void IUnsafe.Magic() {} // looks safe to me
+				public static readonly Safe Instance { get; } = new Safe();
+			}
+
+			private readonly static IUnsafe foo = Safe.Instance;
+		}
+	}";
+
+			// Safe isn't sealed
+			AssertSingleDiagnostic( test, 12, 36, "foo", "test.Tests.Safe" );

```

Comment:
That'd be a bit tricky in general because we'd need to consider control flow (a field declaration could have loops, if statements, multiple returns...)

Not saying it wouldn't be cool to do but it'd definitely be more advanced. A starting point could be to find all return points and consider the union of all their types... but that would even lose things so you'd have to go deeper.